libtransmission/subprocess-posix.cc

```diff
@@ -82,35 +85,33 @@ void set_system_error(tr_error* error, int code, std::string_view what)
 
 [[nodiscard]] bool tr_spawn_async_in_parent(int pipe_fd, tr_error* error)
 {
-    int child_errno = 0;
-    ssize_t count = 0;
+    for (;;)
+    {
+        int child_errno = 0;
+        auto const n_read = read(pipe_fd, &child_errno, sizeof(child_errno));
 
-    static_assert(sizeof(child_errno) == sizeof(errno));
+        if (n_read == -1 && errno == EINTR) // try again
+        {
+            continue;
+        }
 
-    do
-    {
-        count = read(pipe_fd, &child_errno, sizeof(child_errno));
-    } while (count == -1 && errno == EINTR);
+        close(pipe_fd);
```

Comment:
Just a minor nit to reduce indentation. Looks good to me otherwise.

Everything below this line can be moved out of the for loop.

Reply:
that is better. re-revised in 5b3acd3be2b4a73131746473ffe35d1d5aae325f